spice_agent/hardware/actions.py

```diff
@@ -44,6 +47,137 @@ def get_darwin_system_profiler_values(self) -> Dict[str, str]:
             "apple_serial_number": data_type.get("serial_number"),
         }
 
+    def _is_nvidia_smi_available(self) -> bool:
+        """
+        Checks if nvidia-smi is avaialble
+        """
+        try:
+            nvidia_smi_command = "nvidia-smi"
+            subprocess.check_output(nvidia_smi_command)
+            return True
+        except (FileNotFoundError, subprocess.CalledProcessError):
+            return False
+
+    def _get_supported_metal_device(self) -> int:
+        """
+        Checks if metal hardware is supported. If so, the index
+        of the supported metal device is returned
+        """
+        supported_metal_device = -1
+        try:
+            system_profiler_display_data_type_command = (
+                "system_profiler SPDisplaysDataType -json"
+            )
+            system_profiler_display_data_type_output = subprocess.check_output(
+                system_profiler_display_data_type_command.split(" ")
+            )
+            system_profiler_display_data_type_json = json.loads(
+                system_profiler_display_data_type_output
+            )
+
+            # Checks if any attached displays have metal support
+            # Note, other devices here could be AMD GPUs or unconfigured Nvidia GPUs
+            for i, display in enumerate(
+                system_profiler_display_data_type_json["SPDisplaysDataType"]
+            ):
+                if "spdisplays_mtlgpufamilysupport" in display:
+                    supported_metal_device = i
+                    return supported_metal_device
+            return supported_metal_device
+        except (FileNotFoundError, subprocess.CalledProcessError, json.JSONDecodeError):
```

Comment:
you've got 5 different potential things that could fail and rase an exception. we should probably place these try catches around exactly where they could happen instead of globbing them together.

still not sure where the FileNotFoundError would get called